orchestrator/careplanservice/handle_createtask.go

```diff
@@ -91,12 +91,10 @@ func (s *Service) handleCreateTask(ctx context.Context, request FHIRHandlerReque
 			},
 		})
 
-		// Validate Task.For
-		if task.For == nil || task.For.Identifier == nil {
-			return nil, coolfhir.NewErrorWithCode("Task.For must be set with a reference to a patient", http.StatusBadRequest)
-		}
-		if *task.For.Identifier.System != coolfhir.BSNNamingSystem {
-			return nil, coolfhir.NewErrorWithCode("Task.For.Identifier.System must be set with a BSN identifier", http.StatusBadRequest)
+		// Validate Task.For: identifier (with system and value), and/or reference must be set
+		if task.For == nil ||
+			((task.For.Identifier == nil || task.For.Identifier.System == nil || task.For.Identifier.Value == nil) && task.For.Reference == nil) {
```

Comment:
This `if` condition is difficult to parse, wondering if it can be made easier to read. Not blocking though

Reply:
refactored it to a util func